Lab3

Cuperboy opened the pull request:
1) В WPF добавлена панель с вариантами save, load. 2) Добавлено сохранение текущего состояния алгоритма в JSON, а также возможность загрузки этого состояния, и продолжения поиска. 3) Для корректной работы требуются 2 файла. main_data и backup. Это система защиты данных.

Cuperboy committed:
Lab3

1) В WPF добавлена панель с вариантами save, load.
2) Добавлено сохранение текущего состояния алгоритма в JSON, а также возможность загрузки этого состояния, и продолжения поиска.
3) Для корректной работы требуются 2 файла. main_data и backup. Это система защиты данных.

sergey-berezin commented on Wpf/ViewData.cs:
```diff
@@ -10,191 +12,315 @@
+            {
+                string json = JsonConvert.SerializeObject(data_to_json);
+                File.WriteAllText(filename, json);
+                File.WriteAllText(backup, json);
```
Запись в backup может не сработать, если запись в основной файл завершилась с исключением.

sergey-berezin commented on Wpf/MainWindow.xaml.cs:
```diff
@@ -12,65 +12,120 @@
+            Start_button.IsEnabled = true;
+        }
+
+        private void Save_Click(object sender, RoutedEventArgs e)
```
Запись в файл не соответствует требованиям [задания](https://github.com/sergey-berezin/dotnet4/blob/master/courses/autumn2024/Task3.md).

Cuperboy committed:
lab 3 update

0) Для корректной работы нужно поменять string Reestr и string files_root (Reestr - fullpath до базы данных. files_root - path до места, где хранятся эксперименты)

Изменения в программе. Теперь задача выполнена согласно условию задачи.

1) Добавлены 2 окна (выбор имени эксперимента, и выбор экспериментов в реестре)
2) Добавлена обработка событий в случае сбоев и т.д
